core/src/autogluon/core/models/ensemble/bagged_ensemble_model.py

```diff
@@ -622,14 +635,52 @@ def get_features(self):
         assert self.is_fit(), "The model must be fit before calling the get_features method."
         return self.load_child(self.models[0]).get_features()
 
-    def load_child(self, model, verbose=False) -> AbstractModel:
+    def load_child(self, model: Union[AbstractModel, str], verbose=False) -> AbstractModel:
         if isinstance(model, str):
             child_path = self.create_contexts(self.path + model + os.path.sep)
             return self._child_type.load(path=child_path, verbose=verbose)
         else:
             return model
 
-    def save_child(self, model, verbose=False):
+    def add_child(self, model: Union[AbstractModel, str], add_child_times=False, save_child=True, verbose=False):
+        """
+        Add a new fit child model to `self.models`
+
+        Parameters
+        ----------
+        model : Union[AbstractModel, str]
+            The child model to add. If str, it is the name of the model.
+        add_child_times : bool, default = False
+            Whether to add child metadata on times to the bag times.
+            This includes fit_time, predict_time, and predict_1_time.
+        save_child : bool, default = True
+            Whether to save the child model to disk.
+        verbose : bool, default = False
+            Whether to log the child saving process.
+        """
+        if self.models is None:
+            self.models = []
+        if isinstance(model, str):
+            model_name = model
+            model = None
+        else:
+            model_name = model.name
+        if self.low_memory:
+            if save_child:
+                if model is None:
+                    model = self.load_child(model=model_name, verbose=False)
+                self.save_child(model, verbose=verbose)
```

Comment:
[design/naming] From single responsibility principle it's not a good idea to bring save into this method. Saving should be done separately, outside of `add_child` method. Not to mention this is done only in low_memory + save_child section and not in the other ones.

Reply:
saving model and adding model to self.models are linked, because with low_memory setting self.models appends the name, but in low_memory=False, self.models appends the model. We wouldn't want to save the model + add the model to self.models generally speaking.

Do you have a proposal for how it should be done instead?

Reply:
If we passed low_memory to `save_child`, then `save_child` would do nothing when `low_memory=False`. This is probably not what we want.

Reply:
I feel it's ok to do nothing in `save_child` if `low_memory=False`. Basically, save_child will be the function to save the model if needed and not saving it otherwise. @gradientsky what's your thought?

Reply:
> saving model and adding model to self.models are linked, because with low_memory setting self.models appends the name, but in low_memory=False, self.models appends the model. We wouldn't want to save the model + add the model to self.models generally speaking.

They don't need to live in the same method. Separating them reduces cyclomatic complexity of the method and makes unit tests for each of these more concise.

Doing nothing in save_child is ok. The call to `save_child()` can be made after checking `low_memory` and `save_child` outside.

Reply:
I don't necessarily fully agree that it should be separated, but I have done so and it is now revised to address this.